### PR TITLE
Add visual snapshot testing

### LINKED ISSUE #74: [🧪] - Introduce the visual test
https://playwright.dev/docs/test-snapshots

### LINKED ISSUE #75: [🧪] - New test case: Sort functionality
## Desktop
- Given navigate to homepage
- Should show initial data
- When user click header `# Contributions`
- And select sort direction `Asc`
- Then data row should sort by contributions

## Mobile
- Given navigate to homepage
- Should show initial data
- When user click "Sort" button
- And select sort "By Contributions"
- Then data row should sort by contributions

### PR DESCRIPTION
**Pull request by ImBIOS**
Fix #74
Fix #75

### CODE REVIEW

**Comment by ImBIOS** [comment]
wait... it's to tiny to be single PR, I'm attempting to address all the task under the "umbrella"

File: tests/home.spec.ts [hunk 1]
```diff
@@ -36,14 +40,15 @@ test.describe('Homepage', () => {
...
+    async ({ page }) => {
       await test.step('When user search with a valid keyword', async () => {
         await homePage.fillAndSearch('depapp');
       });
 
       await test.step('Then table should show expected result', async () => {
         await expect(homePage.getUsername('depapp')).toBeVisible();
         await expect(homePage.emptyState).not.toBeVisible();
+        await expect(page).toHaveScreenshot('search-results-valid.png', { maxDiffPixelRatio: 0.02 });
       });
```
Comment (mazipan) [inline tests/home.spec.ts:52]: Is it better to add another step to assert the visual?

Something like this:
```suggestion
     await test.step('Then validate visual for search result', async () => {
        await expect(page).toHaveScreenshot('search-results-valid.png', { maxDiffPixelRatio: 0.02 });
     ]})
```

cc: @depapp

**Review by mazipan — CHANGES_REQUESTED** [review]

**Review by depapp — COMMENTED** [review]
thanks for the PR mas @ImBIOS 🙇🏻 

anyway, since this visual testing will use the baseline from the `__screenshots__` directory, and we might have different operating systems (which could affect the appearance of the 🥇🥈🥉 icons), would it be better to ignore the directory?
wdyt mas @ImBIOS & mas @mazipan ?

**Comment by mazipan** [comment]
In the CI, we will generate the baseline snapshot using Ubuntu.
So the approach is usually by running the test inside ubuntu docker for local development.

**Comment by mazipan** [comment]
To appreciate the effort, how if we merge this one.
And I will work on this visual test later.
@depapp

**Comment by depapp** [comment]
> To appreciate the effort, how if we merge this one.
> And I will work on this visual test later.

sure, I agree. approved

**Review by depapp — APPROVED** [review]

**Pull request merged**